From 9df1f57aa1e714cf7b2ce459abf91f1c987f1d76 Mon Sep 17 00:00:00 2001
From: root <root@lr70-1-poplar-12.openstack-dev.graphcore.ai>
Date: Mon, 5 Dec 2022 20:10:56 +0000
Subject: [PATCH 3/5] Fix an error

---
 freqtrade/exchange/exchange.py | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/freqtrade/exchange/exchange.py b/freqtrade/exchange/exchange.py
index 32d4f3435..f556beeb4 100644
--- a/freqtrade/exchange/exchange.py
+++ b/freqtrade/exchange/exchange.py
@@ -482,15 +482,7 @@ class Exchange:
 
     def _load_markets(self) -> None:
         """ Initialize markets both sync and async """
-        try:
-            self._markets = self._api.load_markets()
-            self._load_async_markets()
-            self._last_markets_refresh = arrow.utcnow().int_timestamp
-            if self._ft_has['needs_trading_fees']:
-                self._trading_fees = self.fetch_trading_fees()
-
-        except ccxt.BaseError:
-            logger.exception('Unable to initialize markets.')
+        pass
 
     def reload_markets(self) -> None:
         """Reload markets both sync and async if refresh interval has passed """
-- 
2.25.1

